### DISCUSSION TITLE
"Cannot return null for non-nullable field"

### DISCUSSION THREAD

**Discussion by hatchli** [prisma/prisma#2030] [Q&A] [answered]
Whenever I try and run the following mutation in the playground:
```
query emailConfirmed($email: String!){
  emailConfirmed(email: $email){
    email
  }
}
```
With the variables:
```
{
  "email": "correctemail@gmail.com"
}
```
I get the following response:
```
{
  "data": null,
  "errors": [
    {
      "message": "Cannot return null for non-nullable field Proposal.email.",
      "locations": [
        {
          "line": 3,
          "column": 5
        }
      ],
      "path": [
        "emailConfirmed",
        "email"
      ]
    }
  ],
  "extensions": {}
}
```

My model:
```
model Proposal {
  created_at         DateTime @default(now())
  email              String   @unique
  emailConfirmToken  String?  @unique
  emailConfirmed     Boolean  @default(false)
  confirmTokenExpiry Float?
  name               String?
  proposal_id        Int      @default(autoincrement()) @id
  type               Type[]
}
```

My mutation:
```
    t.field('emailConfirmed', {
      type: 'Proposal',
      args: {
        email: stringArg({ nullable: false }),
      },
      resolve: (parent, { email }, ctx) => {
        return ctx.prisma.proposal.findOne({
          where: {
            email: email,
          },
          select: {
            emailConfirmed: true,
          },
        })
      },
    })
```

I am confused as to why I cannot query based on one of the unique fields (email). What am I doing wrong exactly?

**Comment by pantharshit00** [comment, marked as answer]
You are only selecting `emailConfirmed` field in your client query while graphql is expecting you to return a full Proposal object.

Try the following:
```ts
 t.field('emailConfirmed', {
      type: 'Proposal',
      args: {
        email: stringArg({ nullable: false }),
      },
      resolve: (parent, { email }, ctx) => {
        return ctx.prisma.proposal.findOne({
          where: {
            email: email,
          },
        })
      },
    })
```

**Reply by nikolasburk** [reply]
Hey @hatchli, did this resolve your issue? If so, do you mind marking this answer as accepted by clicking the little checkmark? 🙂

**Reply by hatchli** [reply]
Thanks @nikolasburk and @pantharshit00 !

**Comment by racso-san** [comment]
Hola necesito ayuda, estoy teniendo el mismo problema y nose como resolverlo
My entity:
------------------------------------------------------------------------------------------------------------------------------------------------------------------------
@ObjectType()
@Entity()
export class User extends BaseEntity {

    @Field()
    @PrimaryGeneratedColumn()
    id!: number;

    @Field()
    @Column()
    name!: String;

    @Field()
    @Column()
    email!: String;

    @Field()
    @Column()
    password!: String;

    @BeforeInsert()
    async beforeInsert() {
      if(this.password === undefined)return; // Si no hay pass no encripta nada
      try{
        this.password = await bcrypt.hash(this.password, 10);
      } catch(err){
            console.log(err);
            throw err;
      }
    }
    
} 
------------------------------------------------------------------------------------------------------------------------------------------------------------------------
Mi resolver:
    @Mutation(() => User, {nullable:true})
    async logIn(
        @Arg("email") email:string) {
            const user = await User.findOne(email);
            return user;
    }
    
}

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
